ping/ping.c

```diff
@@ -787,8 +787,15 @@ int ping4_run(struct ping_rts *rts, int argc, char **argv, struct addrinfo *ai,
 
 			memcpy(&rts->whereto, result->ai_addr, sizeof rts->whereto);
 			memset(hnamebuf, 0, sizeof hnamebuf);
-			if (result->ai_canonname)
-				strncpy(hnamebuf, result->ai_canonname, sizeof hnamebuf - 1);
+
+			/*
+			 * On certain network setup getaddrinfo() can return empty
+			 * ai_canonname. Instead of printing nothing in "PING"
+			 * line use the target.
+			 */
+			strncpy(hnamebuf, result->ai_canonname ? result->ai_canonname :
+					target, sizeof hnamebuf - 1);
+
```

Comment:
Why can't we do this in the else branch?

```
/*
 * ....
 */
if (result->ai_cannoname)
        strcpy(..., result->ai_cannonname, ...);
else
       strcpy(..., target, ...);
```

Reply:
Fixed, thanks!